src/lambda/commands/deploySamApplication.ts

```diff
@@ -13,137 +13,200 @@ import * as nls from 'vscode-nls'
-import { DefaultValidatingSamCliProcessInvoker } from '../../shared/sam/cli/defaultValidatingSamCliProcessInvoker'
 import { SamCliBuildInvocation } from '../../shared/sam/cli/samCliBuild'
+import { getSamCliContext, SamCliContext } from '../../shared/sam/cli/samCliContext'
 import { SamCliDeployInvocation } from '../../shared/sam/cli/samCliDeploy'
 import { SamCliProcessInvoker } from '../../shared/sam/cli/samCliInvokerUtils'
 import { SamCliPackageInvocation } from '../../shared/sam/cli/samCliPackage'
+import { throwAndNotifyIfInvalid } from '../../shared/sam/cli/samCliValidationUtils'
+import { ChannelLogger } from '../../shared/utilities/vsCodeUtils'
 import { SamDeployWizard, SamDeployWizardResponse } from '../wizards/samDeployWizard'
 
 const localize = nls.loadMessageBundle()
 
+interface DeploySamApplicationParameters {
+    sourceTemplatePath: string
+    deployRootFolder: string
+    profile: string
+    region: string
+    packageBucketName: string
+    destinationStackName: string
+    parameterOverrides: Map<string, string>
+}
+
 export async function deploySamApplication(
     {
-        invoker = new DefaultValidatingSamCliProcessInvoker({}),
+        samCliContext = getSamCliContext(),
+        channelLogger,
         ...restParams
     }: {
-        invoker?: SamCliProcessInvoker
-        outputChannel: vscode.OutputChannel
+        samCliContext?: SamCliContext
+        channelLogger: ChannelLogger,
         regionProvider: RegionProvider
     },
     awsContext: Pick<AwsContext, 'getCredentialProfileName'>
-) {
-    const args: SamDeployWizardResponse | undefined = await new SamDeployWizard(restParams.regionProvider).run()
-    if (!args) {
-        return
+): Promise<void> {
+    try {
+        const profile: string | undefined = awsContext.getCredentialProfileName()
+        if (!profile) {
+            throw new NoActiveCredentialError()
+        }
+
+        throwAndNotifyIfInvalid(await samCliContext.validator.detectValidSamCli())
+
+        const deployWizardResponse: SamDeployWizardResponse | undefined = await new SamDeployWizard(
+            restParams.regionProvider
+        ).run()
+
+        if (!deployWizardResponse) {
+            return
+        }
+
+        const deployParameters: DeploySamApplicationParameters = {
+            deployRootFolder: await makeTemporaryToolkitFolder('samDeploy'),
+            destinationStackName: deployWizardResponse.stackName,
+            packageBucketName: deployWizardResponse.s3Bucket,
+            parameterOverrides: deployWizardResponse.parameterOverrides,
+            profile,
+            region: deployWizardResponse.region,
+            sourceTemplatePath: deployWizardResponse.template.fsPath,
+        }
+
+        const deployApplicationPromise = buildPackageDeploy({
+            deployParameters,
+            channelLogger,
+            invoker: samCliContext.invoker,
+        }).then(async () =>
+            await del(deployParameters.deployRootFolder, {
```

Comment:
Is the explicit delete necessary since we're using a `makeTemporaryToolkitFolder`-created directory? If not, remove this; if it is, add a deletion to the catch statement.

Reply:
A catch statement isn't necessary, nothing is thrown out of `buildPackageDeploy`

Reply:
The delete is here because we delegate to `vscode.window.setStatusBarMessage` and then this function exits.